### DISCUSSION TITLE
Make a state slice reading values from another slice using get() is undefined?

### DISCUSSION THREAD

**Discussion by gremo** [pmndrs/zustand#1421] [General]
- The `settings.productsUI` slice is used to control some **default settings of the UI**
- The `productsFetchParams` is being used for **API calls**. **This is not persisted using persist middleware!!!**

The `productsFetchParams` should return default settings. The slice is updated using `updateProductsFetchParams` function, which will trigger an updated API call. `settings.productsUI` remains untouched.

```json5
{
  settings: {
    productsUI: {
      limit: 10,
      sortField: 'name',
    }
  },
  productsFetchParams: {
    page: 1,
    get limit() {
      return get().settings.productsUI.limit;
    },
    get orderBy() {
      return get().settings.productsUI.sortField;
    },
  },
}
```

The problem I'm facing is that `get().settings.productsUI` is undefined. I don't know if it's my fault or this is a bug.

**Comment by dai-shi** [comment]
seems like you are calling `get()` before finishing creating the store. It's `undefined` until `create()` ends.
